pkgs/applications/networking/xpipe/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv, lib, fetchzip, makeDesktopItem }:
+
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  arch = {
+    x86_64-linux = "x86_64";
+    aarch64-linux = "arm64";
+  }.${system} or throwSystem;
+
+  sha256 = {
+    x86_64-linux = "VmJa0DqnFSzq4wJ+HchguDK4tMeKP/XaE2MidXfootM=";
+    aarch64-linux = "LdXfhZK4n8HqN1u/rKIIZhFk7fREXbTuAzyvvhFR6Lo=";
+  }.${system} or throwSystem;
+
+in stdenv.mkDerivation rec {
+  displayname = "XPipe";
+  pname = "xpipe";
+  version = "1.6.0";
+
+  src = fetchzip {
+    url = "https://github.com/xpipe-io/${pname}/releases/download/${version}/xpipe-portable-linux-${arch}.tar.gz";
+    inherit sha256;
+  };
+
+  desktopItem = makeDesktopItem {
+    categories = [ "Network" ];
+    comment = "The shell connection hub and remote file browser for your entire infrastructure";
+    desktopName = "${displayname}";
+    exec = "/opt/${pname}/cli/bin/xpipe open %U";
+    genericName = "Shell connection hub";
+    icon = "/opt/${pname}/logo.png";
+    name = "${displayname}";
+  };
+
+  installPhase = ''
```

Comment:
Remember `runHook preInstall` and `runHook postInstall`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#new-packages